tests/openssh.rs

```diff
@@ -282,324 +350,345 @@ macro_rules! assert_kind {
+        assert_kind!(failed, io::ErrorKind::WriteZero);
+
+        session.close().await.unwrap();
+    }
 }
 
 #[tokio::test]
 #[cfg_attr(not(ci), ignore)]
 async fn bad_remote_command() {
-    let session = Session::connect(&addr(), KnownHosts::Accept).await.unwrap();
-
-    // a bad remote command should result in a _local_ error.
-    let failed = session
-        .command("no such program")
-        .output()
-        .await
-        .unwrap_err();
-    eprintln!("{:?}", failed);
-    assert!(matches!(failed, Error::Remote(ref e) if e.kind() == io::ErrorKind::NotFound));
-
-    // no matter how you run it
-    let failed = session
-        .command("no such program")
-        .status()
-        .await
-        .unwrap_err();
-    eprintln!("{:?}", failed);
-    assert!(matches!(failed, Error::Remote(ref e) if e.kind() == io::ErrorKind::NotFound));
-
-    // even if you spawn first
-    let mut child = session.command("no such program").spawn().unwrap();
-    let failed = child.wait().await.unwrap_err();
-    eprintln!("{:?}", failed);
-    assert!(matches!(failed, Error::Remote(ref e) if e.kind() == io::ErrorKind::NotFound));
-    child.disconnect().await.unwrap_err();
-
-    // of if you want output
-    let child = session.command("no such program").spawn().unwrap();
-    let failed = child.wait_with_output().await.unwrap_err();
-    eprintln!("{:?}", failed);
-    assert!(matches!(failed, Error::Remote(ref e) if e.kind() == io::ErrorKind::NotFound));
+    for session in connects().await {
+        // a bad remote command should result in a _local_ error.
+        let failed = session
+            .command("no such program")
+            .output()
+            .await
+            .unwrap_err();
+        eprintln!("{:?}", failed);
+        assert_kind!(failed, io::ErrorKind::NotFound);
+
+        // no matter how you run it
+        let failed = session
+            .command("no such program")
+            .status()
+            .await
+            .unwrap_err();
+        eprintln!("{:?}", failed);
+        assert_kind!(failed, io::ErrorKind::NotFound);
+
+        // even if you spawn first
+        let mut child = session.command("no such program").spawn().await.unwrap();
+        let failed = child.wait().await.unwrap_err();
+        eprintln!("{:?}", failed);
+        assert_kind!(failed, io::ErrorKind::NotFound);
+        child.disconnect().await.unwrap_err();
+
+        // of if you want output
+        let child = session.command("no such program").spawn().await.unwrap();
+        let failed = child.wait_with_output().await.unwrap_err();
+        eprintln!("{:?}", failed);
+        assert_kind!(failed, io::ErrorKind::NotFound);
+
+        // no matter how hard you _try_
+        let mut child = session.command("no such program").spawn().await.unwrap();
+        std::thread::sleep(std::time::Duration::from_millis(500));
+
+        let res = child.try_wait();
+        if let Ok(None) = res {
+            continue;
+        }
```

Comment:
I want this to be stronger. Specifically, I want this test to start failing once we fix `try_wait` for the native mux backend. And I don't want to allow the process mux backend to return `Ok(None)`. I think the easiest way to deal with that is to change the `for` loop for this test to use `.enumerate()` and then to check on the index here.

Reply:
OK, I've fixed this by removing this `if`.